### PR TITLE
feat: get-user-by-id

### PR DESCRIPTION
**Pull request by khayss**
## What does the PR do?
This PR is about getting a user by Id

## How should this be manually tested?
Send a GET request to `/api/v1/users/:userId` with access token in the authorization header.

## Checklist of what you did
- [x] Implemented endpoint to get a user by the `user_id`.
- [x] Handle error for invalid user id in the params.
- [x] Integrate with auth guard to prevent request from unauthorized access.
- [x] Implement safety measure to avoid leak of user sensitive data such as password.
- [x] Requests to the endpoint must include a valid authentication token in the Authorization header.
- [x] Returned success `status_code ` with the user data.
- [x] Unit test the user service to ensure it works as expected.

## Test Screenshot

image

### CODE REVIEW

**Review by Heba-WebDev — DISMISSED** [review]

File: src/modules/user/user.controller.ts [hunk 1]
```diff
@@ -45,4 +45,14 @@ export class UserController {
       message: result.message,
     };
   }
+
+  @ApiBearerAuth()
+  @ApiOperation({ summary: 'Get User Data' })
+  @ApiResponse({ status: 200, description: 'User data fetched successfully' })
+  @ApiResponse({ status: 401, description: 'Unauthorized' })
+  @ApiResponse({ status: 500, description: 'Internal Server Error' })
+  @Get(':userId')
```
Comment (Cobidex) [inline src/modules/user/user.controller.ts:54]: This should be id not userId in compliance with restful API convention

File: src/modules/user/user.controller.ts [hunk 2]
```diff
@@ -45,4 +45,14 @@ export class UserController {
...
+  async getUserDataById(@Param('userId') userId: string) {
```
Comment (Cobidex) [inline src/modules/user/user.controller.ts:55]: Same here

**Comment by khayss** [comment]
made the adjustments @Cobidex

**Review by Cobidex — APPROVED** [review]
LGTM

**Comment by CFCIfe** [comment]
Sorry, what issue is this fixing?

also your branch name is not following convention.

**Pull request closed without merging**